### PR TITLE
MMDLoader Image Cross-origin issue solution

### PR DESCRIPTION
**Pull request by takahirox**
`getImageData()` of `CanvasRenderingContext2D` with image in other domain
for transparency detection fails because of cross-origin problem
even if server responds with "Access-Control-Allow-Origin: *".

If `MMDLoader.imageCrossOrigin` is true, `MMDLoader` tries to solve this issue with

    var image = new Image();
    image.crossOrigin = 'anonymous';
    image.src = url;

### CODE REVIEW

File: examples/js/loaders/MMDLoader.js [hunk 1]
```diff
@@ -67,6 +68,23 @@ THREE.MMDLoader.prototype.defaultToonTextures = [
...
+/*
+ * getImageData() of CanvasRenderingContext2D with image in other domain
+ * for transparency detection fails because of cross-origin problem
+ * even if server responds with "Access-Control-Allow-Origin: *".
+ * If this.imageCrossOrigin is true, MMDLoader tries to solve this issue
+ * with
+ *   var image = new Image();
+ *   image.crossOrigin = 'anonymous';
+ *   image.src = url;
+ * This workaround can be removed if ImageLoader supports crossOrigin.
```
Comment (takahirox) [inline examples/js/loaders/MMDLoader.js]: `ImageLoader.crossOrigin` isn't used for anything now

https://github.com/mrdoob/three.js/blob/c2cec2e7dca8b6b47ce25f831635b13275e0217f/src/loaders/ImageLoader.js#L58

Even if I set `image.crossOrigin = 'anonymous';` right before `image.src = URL.createObjectURL( blob );`, we can't avoid cross-origin error on `getImageData()` of `CanvasRenderingContext2D`.

https://github.com/mrdoob/three.js/blob/c2cec2e7dca8b6b47ce25f831635b13275e0217f/src/loaders/ImageLoader.js#L46

It seems like `image.crossOrigin = 'anonymous';` won't work for ``image.src = URL.createObjectURL( blob );`. I speculate cross-origin info is decided at the `URL.createObjectURL` or `XMLHttpRequest.send` moment and the info will be just propagated to `image`. Probably `image.crossOrigin = 'anonymous';` can't block this propagation.

Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: I see...  I wonder if the new `ImageBitmap` API solves this...

https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap

Comment (takahirox) [inline examples/js/loaders/MMDLoader.js]: This new API seems avoiding the cross origin issue but I've faced

three.js:19115 DOMException: Failed to execute 'texImage2D' on 'WebGLRenderingContext': The cross-origin image at blob:null/8074b93f-ff43-46be-9339-e81f20a88114 may not be loaded.
texImage2D @ three.js:19115
uploadTexture @ three.js:17989
setTexture2D @ three.js:17667
setTexture2D @ three.js:22009
setValueT1 @ three.js:4596
WebGLUniforms.upload @ three.js:4923
setProgram @ three.js:21448
WebGLRenderer.renderBufferDirect @ three.js:20224
renderObjects @ three.js:21015
WebGLRenderer.render @ three.js:20752
render @ OutlineEffect.js:441
render @ MMDLoader.js:2748
render @ webgl_loader_mmd3.html:247
animate @ webgl_loader_mmd3.html:235

I'll see the detail later...

Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: https://twitter.com/stephomi/status/799734648637190144

I guess we can ignore `ImageBitmap` for now... 😕

Comment (mrdoob) [inline examples/js/loaders/MMDLoader.js]: @takahirox That code looks good. I'm also considering reverting to not using XHRLoader/FileLoader internally...

Comment (takahirox) [inline examples/js/loaders/MMDLoader.js]: Cool, I mage another PR #10168
And I'll close this  PR.

**Pull request closed without merging**